Docker (#209)
dyc3/opentogethertube: docker/DOCKER.md

## Deploy Docker image to the Docker hub

Use the production Dockerfile that is optimized for a smaller size image for Docker Hub

To use the create the deploy image, please follow the steps below.

1. Next you need to set up your configuration. Start by copying the example
   config in the `env` folder to a new file called `example.env`

```bash
cp env/example.env env/production.env
```

2. Open `env/production.env` and make sure the file is empty to not leak any api keys.

3. To build the image, make sure you are in the project root `/opentogethertube`.

```bash
DOCKER_HUB_USERNAME=yourhubusername
PACKAGE_VERSION=$(node -p "require('./package.json').version")
docker build -f docker/prod/Dockerfile -t $DOCKER_HUB_USERNAME/opentogethertube:$PACKAGE_VERSION .
```

4. Login to Docker hub and push the docker image

```bash
docker login --username=$DOCKER_HUB_USERNAME # enter your password
docker push $DOCKER_HUB_USERNAME/opentogethertube:$PACKAGE_VERSION
```

### Setup Travis CI

1. Install travis cli tool to add secret env to the ci

```bash
sudo gem install travis
travis login
travis encrypt DOCKER_HUB_USERNAME=<email> --add
travis encrypt DOCKER_HUB_PASSWORD=<password> --add
```

> 2. This will add env to the `.travis.yml` section with the secret password and username for docker hub

Reviewer: Remove these instructions, or move them to a wiki page

Reviewer: And I think some of these commands are wrong

# Docker

[Install Docker](https://docs.docker.com/install/)

You can run the application using the development image or production/deployment image.

## Development

The development image use docker-compose with redis and postgres in seperate containers.

To use the development image, please follow the steps below.

1. Next you need to set up your configuration. Start by copying the example
config in the `env` folder to a new file called `development.env`

   ```bash
   cp env/example.env env/development.env

>    ```

Reviewer: Don't indent these code blocks please

2. Open `env/development.env` and replace `API_KEY_GOES_HERE` with the youtube api key.

3. To build the image locally, make sure you are in the project root `/opentogethertube`
   before runing the command below.

   ```bash

>    docker-compose -f docker/prod/docker-compose.yml up -d

Reviewer: ```suggestion
   docker-compose -f docker/dev/docker-compose.yml up -d
```

   ```

4. Go to http://localhost:3001

> You can run the unit test from the dev container using this command.

Reviewer: ```suggestion
You can run the test suite from the dev container using this command.
```

```bash
docker exec -it opentogethertube-dev npm test
```

You can run the linter from the dev container using this command.

```bash
docker exec -it opentogethertube-dev npm lint
```

## Production

> The production image use docker-compose with redis and postgres in seperate containers.

Reviewer: ```suggestion
The production image uses docker-compose with redis and postgres in separate containers.
```
